src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls

```diff
@@ -0,0 +1,279 @@
+        timeoutCount++;
+
+        refreshCustomSettings();
+        setDeploymentStatus();
+
+        if (deploymentResult != null) {
+            // We had a migration that contained CMT records to insert
+            Boolean isSuccess = deploymentResult.status == Metadata.DeployStatus.Succeeded
+                && dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING;
+
+            if (isSuccess) {
+                isPolling = false;
+                throwMessage(ApexPages.Severity.CONFIRM, Label.stgDeploymentCustomMetadataMappingEnabled + ' ' +
+                    migrationMappingUtility.queuedMetadataTypesForDeploy.size() + ' custom metadata records were deployed.');
+            } else if (deploymentResult.status == Metadata.DeployStatus.FAILED) {
+                isPolling = false;
+                throwMessage(ApexPages.Severity.ERROR, Label.stgDeploymentTryAgain);
+            }
+
+        } else if (deploymentResult == null
+                   && dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING) {
+            // We had a migration that didn't contain CMT records to insert
+            // DeploymentContainer was empty which means we're using our standard field mapping set
+            // Org had no conflicting field mappings nor any net new field mappings in their Data Import help text
+            isPolling = false;
+            throwMessage(ApexPages.Severity.CONFIRM, Label.stgDeploymentCustomMetadataMappingEnabled);
+        }
+
+        if (timeoutCount > MAX_TIME_OUT) {
+            isPolling = false;
+            throwMessage(ApexPages.Severity.WARNING, Label.stgDeploymentTimedOut + ' ' + Label.stgDeploymentTryAgain);
+            return null;
+        }
+
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description Checks to see if we have a saved custom metadata deployment id and sets deploymentResult
+    */
+    private void setDeploymentStatus() {
+        if (dataImportSettings.CMT_Deployment_ID__c != null) {
+            deploymentResult =
+                CMT_MetadataAPI.getDeploymentResult(dataImportSettings.CMT_Deployment_ID__c);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Checks to see if we have a saved custom metadata deployment id,
+    * inspects the deployment status, and sets the hasDeploymentInProgress boolean
+    */
+    private void refreshCustomSettings() {
+        dataImportSettings = UTIL_CustomSettingsFacade.getDataImportSettings();
+    }
+
+    /*******************************************************************************************************
+    * @description Wrapper for Metadata.DeployResult
+    */
+    private void throwMessage(ApexPages.Severity severity, String message) {
```

Comment:
How about `addPageMessage` rather than throwMessage since it is not actually throwing an exception?